lib/arel/select_manager.rb

```diff
@@ -6,11 +6,15 @@ class SelectManager < Arel::TreeManager
 
     STRING_OR_SYMBOL_CLASS = [Symbol, String]
 
-    def initialize engine, table = nil
+    def initialize engine, table = nil, cores = nil
       super(engine)
-      @ast   = Nodes::SelectStatement.new
-      @ctx    = @ast.cores.last
-      from table
+      if cores
+        @ast = Nodes::SelectStatement.new cores
```

Comment:
Actually, I did it before, but ended up skipping it when I rebased when it caused a conflict and I decided it was pretty trivial. I guess if people feel strongly, I can add it back.